pkg/ble/device/twindata.go

```diff
@@ -0,0 +1,127 @@
+package device
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+
+	"github.com/currantlabs/ble"
+	"github.com/kubeedge/kubeedge/cloud/pkg/apis/devices/v1alpha2"
+	"github.com/kubeedge/mappers-go/pkg/ble/configmap"
+	"github.com/kubeedge/mappers-go/pkg/ble/driver"
+	"github.com/kubeedge/mappers-go/pkg/ble/globals"
+	"github.com/kubeedge/mappers-go/pkg/common"
+	"k8s.io/klog/v2"
+)
+
+// TwinData is the timer structure for getting twin/data.
+type TwinData struct {
+	BluetoothClient        *driver.BluetoothClient
+	Name                   string
+	Type                   string
+	BluetoothVisitorConfig configmap.BluetoothVisitorConfig
+	Result                 string
+	Topic                  string
+}
+
+// Run timer function.
+func (td *TwinData) Run() {
+	uuid := ble.MustParse(td.BluetoothVisitorConfig.CharacteristicUUID)
+	if p, err := td.BluetoothClient.Client.DiscoverProfile(true); err == nil {
+		if u := p.Find(ble.NewCharacteristic(uuid)); u != nil {
+			c := u.(*ble.Characteristic)
+			// read data actively
+			if (c.Property & ble.CharRead) != 0 {
+				b, err := td.BluetoothClient.Client.ReadCharacteristic(c)
```

Comment:
It's better to pack it to a read function that aligned with write.

Reply:
/lgtm, it had been fixed